Installing nassl fails (pshtt dep) in vagrant using docker

Up to date Ubuntu 16.04.1 bare server install.  Installed vagrant, ansible, docker.io

TASK [securethenews-deploy : Install pshtt] ************************************
fatal: [securethenews]: FAILED! => {"changed": false, "cmd": "/usr/local/bin/pip install -e git+git://github.com/dhs-ncats/pshtt@7362a8bdabe7190a933e4c5e18561ba9d07dc561#egg=pshtt", "failed": true, "msg": "stdout: Obtaining pshtt from git+git://github.com/dhs-ncats/pshtt@7362a8bdabe7190a933e4c5e18561ba9d07dc561#egg=pshtt\n  Cloning git://github.com/dhs-ncats/pshtt (to 7362a8bdabe7190a933e4c5e18561ba9d07dc561) to ./src/pshtt\nRequirement already satisfied: requests>=2.10.0 in /usr/local/lib/python3.4/dist-packages (from pshtt)\nCollecting sslyze>=0.13.6 (from pshtt)\n  Downloading SSLyze-0.14.2.tar.gz (1.1MB)\nCollecting wget>=3.2 (from pshtt)\n  Downloading wget-3.2.zip\nCollecting docopt (from pshtt)\n  Downloading docopt-0.6.2.tar.gz\nCollecting requests_cache (from pshtt)\n  Downloading requests_cache-0.4.13-py2.py3-none-any.whl\nCollecting nassl<0.15.0,>=0.14.0 (from sslyze>=0.13.6->pshtt)\n  Downloading nassl-0.14.2.tar.gz (15.2MB)\n    Complete output from command python setup.py egg_info:\n    Traceback (most recent call last):\n      File \"<string>\", line 1, in <module>\n      File \"/tmp/pip-build-um741sgh/nassl/setup.py\", line 8, in <module>\n        from nassl import __author__, __version__\n      File \"/tmp/pip-build-um741sgh/nassl/nassl/__init__.py\", line 24\n        SSL_MODE_SEND_FALLBACK_SCSV =       0x00000080L\n                                                      ^\n    SyntaxError: invalid syntax\n    \n    ----------------------------------------\n\n:stderr:   Could not find a tag or branch '7362a8bdabe7190a933e4c5e18561ba9d07dc561', assuming commit.\nCommand \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-um741sgh/nassl/\n"}

PLAY RECAP *********************************************************************
securethenews              : ok=17   changed=12   unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
Makefile:8: recipe for target 'dev' failed
make: *** [dev] Error 1


Full log:

steve@ubuntu-dev:~/securethenews$ make dev
make ansible
make[1]: Entering directory '/home/steve/securethenews'
ansible-galaxy install -r ansible/requirements.yml -p ansible/roles
- geerlingguy.nodejs is already installed, skipping.
make[1]: Leaving directory '/home/steve/securethenews'
vagrant up --provision
Bringing machine 'securethenews' up with 'docker' provider...
==> securethenews: Building the container from a Dockerfile...
    securethenews: Sending build context to Docker daemon 2.048 kB
    securethenews: Step 1 : FROM jesselang/debian-vagrant:jessie
    securethenews: jessie: Pulling from jesselang/debian-vagrant
    securethenews: 5040bd298390: Pulling fs layer
    securethenews: 2267135e1e07: Pulling fs layer
    securethenews: 8b6c5f74664e: Pulling fs layer
    securethenews: 9bdc99613608: Pulling fs layer
    securethenews: 8d31f1db809a: Pulling fs layer
    securethenews: 3a35cc614636: Pulling fs layer
    securethenews: d7cf751a7af0: Pulling fs layer
    securethenews: cf074cfdd2e2: Pulling fs layer
    securethenews: 6c7158c01a4d: Pulling fs layer
    securethenews: e554a7e3ebaf: Pulling fs layer
    securethenews: 9844c47a3dc7: Pulling fs layer
    securethenews: b0536c63952a: Pulling fs layer
    securethenews: e7bfcad7fb37: Pulling fs layer
    securethenews: e02d59f474b9: Pulling fs layer
    securethenews: 862ea11a5693: Pulling fs layer
    securethenews: 77e9570dbfe8: Pulling fs layer
    securethenews: 9bdc99613608: Waiting
    securethenews: 8d31f1db809a: Waiting
    securethenews: 3a35cc614636: Waiting
    securethenews: d7cf751a7af0: Waiting
    securethenews: cf074cfdd2e2: Waiting
    securethenews: 6c7158c01a4d: Waiting
    securethenews: e554a7e3ebaf: Waiting
    securethenews: 9844c47a3dc7: Waiting
    securethenews: b0536c63952a: Waiting
    securethenews: e7bfcad7fb37: Waiting
    securethenews: e02d59f474b9: Waiting
    securethenews: 862ea11a5693: Waiting
    securethenews: 77e9570dbfe8: Waiting
    securethenews: 8b6c5f74664e:
    securethenews: Verifying Checksum
    securethenews: 8b6c5f74664e:
    securethenews: Download complete
    securethenews: 9bdc99613608: Verifying Checksum
    securethenews: 9bdc99613608: Download complete
    securethenews: 2267135e1e07: Verifying Checksum
    securethenews: 2267135e1e07: Download complete
    securethenews: 8d31f1db809a: Verifying Checksum
    securethenews: 8d31f1db809a: Download complete
    securethenews: 3a35cc614636: Verifying Checksum
    securethenews: 3a35cc614636: Download complete
    securethenews: d7cf751a7af0: Verifying Checksum
    securethenews: d7cf751a7af0: Download complete
    securethenews: cf074cfdd2e2: Verifying Checksum
    securethenews: cf074cfdd2e2: Download complete
    securethenews: 6c7158c01a4d: Verifying Checksum
    securethenews: 6c7158c01a4d: Download complete
    securethenews: e554a7e3ebaf: Verifying Checksum
    securethenews: e554a7e3ebaf: Download complete
    securethenews: 9844c47a3dc7: Verifying Checksum
    securethenews: 9844c47a3dc7: Download complete
    securethenews: b0536c63952a: Verifying Checksum
    securethenews: b0536c63952a: Download complete
    securethenews: e7bfcad7fb37: Verifying Checksum
    securethenews: e7bfcad7fb37: Download complete
    securethenews: e02d59f474b9: Verifying Checksum
    securethenews: e02d59f474b9: Download complete
    securethenews: 862ea11a5693: Verifying Checksum
    securethenews: 862ea11a5693: Download complete
    securethenews: 77e9570dbfe8: Verifying Checksum
    securethenews: 77e9570dbfe8: Download complete
    securethenews: 5040bd298390: Verifying Checksum
    securethenews: 5040bd298390: Download complete
    securethenews: 5040bd298390:
    securethenews: Pull complete
    securethenews: 2267135e1e07:
    securethenews: Pull complete
    securethenews: 8b6c5f74664e:
    securethenews: Pull complete
    securethenews: 9bdc99613608:
    securethenews: Pull complete
    securethenews: 8d31f1db809a: Pull complete
    securethenews: 3a35cc614636: Pull complete
    securethenews: d7cf751a7af0: Pull complete
    securethenews: cf074cfdd2e2: Pull complete
    securethenews: 6c7158c01a4d: Pull complete
    securethenews: e554a7e3ebaf: Pull complete
    securethenews: 9844c47a3dc7: Pull complete
    securethenews: b0536c63952a: Pull complete
    securethenews: e7bfcad7fb37: Pull complete
    securethenews: e02d59f474b9: Pull complete
    securethenews: 862ea11a5693: Pull complete
    securethenews: 77e9570dbfe8: Pull complete
    securethenews: Digest: sha256:d13dbf0e978d10ae7c4d25c1647cccc4bea2d1c2c2bd4969754e6eaf8b83a814
    securethenews: Status: Downloaded newer image for jesselang/debian-vagrant:jessie
    securethenews:  ---> d2da839b9175
    securethenews: Step 2 : RUN apt-get install --no-install-recommends -y python
    securethenews:  ---> Running in ba4dd43361a1
    securethenews: Reading package lists...
    securethenews: Building dependency tree...
    securethenews:
    securethenews: Reading state information...
    securethenews: The following extra packages will be installed:
    securethenews:   libexpat1 libffi6 libpython-stdlib libpython2.7-minimal libpython2.7-stdlib
    securethenews:   libsqlite3-0 mime-support python-minimal python2.7 python2.7-minimal
    securethenews: Suggested packages:
    securethenews:   python-doc python-tk python2.7-doc binutils binfmt-support
    securethenews: Recommended packages:
    securethenews:   file
    securethenews: The following NEW packages will be installed:
    securethenews:   libexpat1 libffi6 libpython-stdlib libpython2.7-minimal libpython2.7-stdlib
    securethenews:   libsqlite3-0 mime-support python python-minimal python2.7 python2.7-minimal
    securethenews: 0 upgraded, 11 newly installed, 0 to remove and 0 not upgraded.
    securethenews: Need to get 4663 kB of archives.
    securethenews: After this operation, 18.0 MB of additional disk space will be used.
    securethenews: Get:1 http://deb.debian.org/debian/ jessie/main libsqlite3-0 amd64 3.8.7.1-1+deb8u2 [438 kB]
    securethenews: Get:2 http://deb.debian.org/debian/ jessie/main libpython2.7-minimal amd64 2.7.9-2+deb8u1 [378 kB]
    securethenews: Get:3 http://deb.debian.org/debian/ jessie/main python2.7-minimal amd64 2.7.9-2+deb8u1 [1401 kB]
    securethenews: Get:4 http://deb.debian.org/debian/ jessie/main python-minimal amd64 2.7.9-1 [40.3 kB]
    securethenews: Get:5 http://deb.debian.org/debian/ jessie/main mime-support all 3.58 [36.0 kB]
    securethenews: Get:6 http://deb.debian.org/debian/ jessie/main libexpat1 amd64 2.1.0-6+deb8u3 [80.0 kB]
    securethenews: Get:7 http://deb.debian.org/debian/ jessie/main libffi6 amd64 3.1-2+b2 [20.1 kB]
    securethenews: Get:8 http://deb.debian.org/debian/ jessie/main libpython2.7-stdlib amd64 2.7.9-2+deb8u1 [1847 kB]
    securethenews: Get:9 http://deb.debian.org/debian/ jessie/main python2.7 amd64 2.7.9-2+deb8u1 [252 kB]
    securethenews: Get:10 http://deb.debian.org/debian/ jessie/main libpython-stdlib amd64 2.7.9-1 [19.5 kB]
    securethenews: Get:11 http://deb.debian.org/debian/ jessie/main python amd64 2.7.9-1 [151 kB]
    securethenews: debconf: delaying package configuration, since apt-utils is not installed
    securethenews:
    securethenews: Fetched 4663 kB in 3s (1238 kB/s)
    securethenews: Selecting previously unselected package libsqlite3-0:amd64.
    securethenews: (Reading database ...
    securethenews: (Reading database ... 5%
    securethenews: (Reading database ... 10%
    securethenews: (Reading database ... 15%
    securethenews: (Reading database ... 20%
(Reading database ... 30%ng database ... 25%
(Reading database ... 40%ng database ... 35%
(Reading database ... 50%ng database ... 45%
(Reading database ... 60%ng database ... 55%
(Reading database ... 70%ng database ... 65%
(Reading database ... 80%ng database ... 75%
(Reading database ... 90%ng database ... 85%
(Reading database ... 100%g database ... 95%
    securethenews: (Reading database ... 7814 files and directories currently installed.)
    securethenews: Preparing to unpack .../libsqlite3-0_3.8.7.1-1+deb8u2_amd64.deb ...
    securethenews: Unpacking libsqlite3-0:amd64 (3.8.7.1-1+deb8u2) ...
    securethenews: Selecting previously unselected package libpython2.7-minimal:amd64.
    securethenews: Preparing to unpack .../libpython2.7-minimal_2.7.9-2+deb8u1_amd64.deb ...
    securethenews: Unpacking libpython2.7-minimal:amd64 (2.7.9-2+deb8u1) ...
    securethenews: Selecting previously unselected package python2.7-minimal.
    securethenews: Preparing to unpack .../python2.7-minimal_2.7.9-2+deb8u1_amd64.deb ...
    securethenews: Unpacking python2.7-minimal (2.7.9-2+deb8u1) ...
    securethenews: Selecting previously unselected package python-minimal.
    securethenews: Preparing to unpack .../python-minimal_2.7.9-1_amd64.deb ...
    securethenews: Unpacking python-minimal (2.7.9-1) ...
    securethenews: Selecting previously unselected package mime-support.
    securethenews: Preparing to unpack .../mime-support_3.58_all.deb ...
    securethenews: Unpacking mime-support (3.58) ...
    securethenews: Selecting previously unselected package libexpat1:amd64.
    securethenews: Preparing to unpack .../libexpat1_2.1.0-6+deb8u3_amd64.deb ...
    securethenews: Unpacking libexpat1:amd64 (2.1.0-6+deb8u3) ...
    securethenews: Selecting previously unselected package libffi6:amd64.
    securethenews: Preparing to unpack .../libffi6_3.1-2+b2_amd64.deb ...
    securethenews: Unpacking libffi6:amd64 (3.1-2+b2) ...
    securethenews: Selecting previously unselected package libpython2.7-stdlib:amd64.
    securethenews: Preparing to unpack .../libpython2.7-stdlib_2.7.9-2+deb8u1_amd64.deb ...
    securethenews: Unpacking libpython2.7-stdlib:amd64 (2.7.9-2+deb8u1) ...
    securethenews: Selecting previously unselected package python2.7.
    securethenews: Preparing to unpack .../python2.7_2.7.9-2+deb8u1_amd64.deb ...
    securethenews: Unpacking python2.7 (2.7.9-2+deb8u1) ...
    securethenews: Selecting previously unselected package libpython-stdlib:amd64.
    securethenews: Preparing to unpack .../libpython-stdlib_2.7.9-1_amd64.deb ...
    securethenews: Unpacking libpython-stdlib:amd64 (2.7.9-1) ...
    securethenews: Setting up libpython2.7-minimal:amd64 (2.7.9-2+deb8u1) ...
    securethenews: Setting up python2.7-minimal (2.7.9-2+deb8u1) ...
    securethenews: Linking and byte-compiling packages for runtime python2.7...
    securethenews: Setting up python-minimal (2.7.9-1) ...
    securethenews: Selecting previously unselected package python.
    securethenews: (Reading database ...
(Reading database ... 10%ng database ... 5%
(Reading database ... 20%ng database ... 15%
(Reading database ... 30%ng database ... 25%
(Reading database ... 40%ng database ... 35%
(Reading database ... 50%ng database ... 45%
(Reading database ... 60%ng database ... 55%
(Reading database ... 70%ng database ... 65%
(Reading database ... 80%ng database ... 75%
(Reading database ... 90%ng database ... 85%
(Reading database ... 100%g database ... 95%
    securethenews: (Reading database ... 8610 files and directories currently installed.)
    securethenews: Preparing to unpack .../python_2.7.9-1_amd64.deb ...
    securethenews: Unpacking python (2.7.9-1) ...
    securethenews: Setting up libsqlite3-0:amd64 (3.8.7.1-1+deb8u2) ...
    securethenews: Setting up mime-support (3.58) ...
    securethenews: Setting up libexpat1:amd64 (2.1.0-6+deb8u3) ...
    securethenews: Setting up libffi6:amd64 (3.1-2+b2) ...
    securethenews: Setting up libpython2.7-stdlib:amd64 (2.7.9-2+deb8u1) ...
    securethenews: Setting up python2.7 (2.7.9-2+deb8u1) ...
    securethenews: Setting up libpython-stdlib:amd64 (2.7.9-1) ...
    securethenews: Setting up python (2.7.9-1) ...
    securethenews: Processing triggers for libc-bin (2.19-18+deb8u7) ...
    securethenews:  ---> b6a4dc08fad9
    securethenews: Removing intermediate container ba4dd43361a1
    securethenews: Successfully built b6a4dc08fad9
    securethenews:
    securethenews: Image: b6a4dc08fad9
==> securethenews: Creating the container...
    securethenews:   Name: securethenews_securethenews_1486162108
    securethenews:  Image: b6a4dc08fad9
    securethenews: Volume: /home/steve/securethenews:/vagrant
    securethenews:   Port: 8000:8000
    securethenews:   Port: 35729:35729
    securethenews:   Port: 127.0.0.1:2222:22
    securethenews:
    securethenews: Container created: 036d4b38da609472
==> securethenews: Starting container...
==> securethenews: Waiting for machine to boot. This may take a few minutes...
    securethenews: SSH address: 172.17.0.2:22
    securethenews: SSH username: vagrant
    securethenews: SSH auth method: private key
    securethenews:
    securethenews: Vagrant insecure key detected. Vagrant will automatically replace
    securethenews: this with a newly generated keypair for better security.
    securethenews:
    securethenews: Inserting generated public key within guest...
    securethenews: Removing insecure key from the guest if it's present...
    securethenews: Key inserted! Disconnecting and reconnecting using new SSH key...
==> securethenews: Machine booted and ready!
==> securethenews: Running provisioner: ansible...
    securethenews: Running ansible-playbook...

PLAY [Deploy Secure The News development machine] ******************************

TASK [setup] *******************************************************************
ok: [securethenews]

TASK [geerlingguy.nodejs : include] ********************************************
skipping: [securethenews]

TASK [geerlingguy.nodejs : include] ********************************************
included: /home/steve/securethenews/ansible/roles/geerlingguy.nodejs/tasks/setup-Debian.yml for securethenews

TASK [geerlingguy.nodejs : Ensure apt-transport-https is installed.] ***********
changed: [securethenews]

TASK [geerlingguy.nodejs : Add Nodesource apt key.] ****************************
changed: [securethenews]

TASK [geerlingguy.nodejs : Add NodeSource repositories for Node.js.] ***********
changed: [securethenews] => (item=deb https://deb.nodesource.com/node_4.x jessie main)
changed: [securethenews] => (item=deb-src https://deb.nodesource.com/node_4.x jessie main)

TASK [geerlingguy.nodejs : Update apt cache if repo was added.] ****************
ok: [securethenews]

TASK [geerlingguy.nodejs : Ensure Node.js and npm are installed.] **************
changed: [securethenews]

TASK [geerlingguy.nodejs : Define nodejs_install_npm_user] *********************
ok: [securethenews]

TASK [geerlingguy.nodejs : Create npm global directory] ************************
changed: [securethenews]

TASK [geerlingguy.nodejs : Add npm_config_prefix bin directory to global $PATH.] ***
changed: [securethenews]

TASK [geerlingguy.nodejs : Ensure npm global packages are installed.] **********

TASK [geerlingguy.nodejs : Ensure npm global packages are at the latest release.] ***

TASK [securethenews-deploy : Install apt dependencies.] ************************
changed: [securethenews] => (item=[u'aptitude', u'git', u'ipython3', u'libjpeg-dev', u'postgresql-server-dev-all', u'python-dev', u'python-pip', u'python3-pip', u'ruby', u'tmux', u'vim', u'zlib1g-dev'])

TASK [securethenews-deploy : Upgrade all packages.] ****************************
ok: [securethenews]

TASK [securethenews-deploy : Upgrade pip for Python 2 to avoid ImportError] ****
changed: [securethenews]

TASK [securethenews-deploy : Upgrade pip for Python 3 to avoid ImportError] ****
changed: [securethenews]

TASK [securethenews-deploy : Install Django project requirements] **************
changed: [securethenews]

TASK [securethenews-deploy : Install Node dependencies] ************************
changed: [securethenews]

TASK [securethenews-deploy : Install gulp globally] ****************************
changed: [securethenews]

TASK [securethenews-deploy : Install pshtt] ************************************
fatal: [securethenews]: FAILED! => {"changed": false, "cmd": "/usr/local/bin/pip install -e git+git://github.com/dhs-ncats/pshtt@7362a8bdabe7190a933e4c5e18561ba9d07dc561#egg=pshtt", "failed": true, "msg": "stdout: Obtaining pshtt from git+git://github.com/dhs-ncats/pshtt@7362a8bdabe7190a933e4c5e18561ba9d07dc561#egg=pshtt\n  Cloning git://github.com/dhs-ncats/pshtt (to 7362a8bdabe7190a933e4c5e18561ba9d07dc561) to ./src/pshtt\nRequirement already satisfied: requests>=2.10.0 in /usr/local/lib/python3.4/dist-packages (from pshtt)\nCollecting sslyze>=0.13.6 (from pshtt)\n  Downloading SSLyze-0.14.2.tar.gz (1.1MB)\nCollecting wget>=3.2 (from pshtt)\n  Downloading wget-3.2.zip\nCollecting docopt (from pshtt)\n  Downloading docopt-0.6.2.tar.gz\nCollecting requests_cache (from pshtt)\n  Downloading requests_cache-0.4.13-py2.py3-none-any.whl\nCollecting nassl<0.15.0,>=0.14.0 (from sslyze>=0.13.6->pshtt)\n  Downloading nassl-0.14.2.tar.gz (15.2MB)\n    Complete output from command python setup.py egg_info:\n    Traceback (most recent call last):\n      File \"<string>\", line 1, in <module>\n      File \"/tmp/pip-build-um741sgh/nassl/setup.py\", line 8, in <module>\n        from nassl import __author__, __version__\n      File \"/tmp/pip-build-um741sgh/nassl/nassl/__init__.py\", line 24\n        SSL_MODE_SEND_FALLBACK_SCSV =       0x00000080L\n                                                      ^\n    SyntaxError: invalid syntax\n    \n    ----------------------------------------\n\n:stderr:   Could not find a tag or branch '7362a8bdabe7190a933e4c5e18561ba9d07dc561', assuming commit.\nCommand \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-um741sgh/nassl/\n"}

PLAY RECAP *********************************************************************
securethenews              : ok=17   changed=12   unreachable=0    failed=1

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
Makefile:8: recipe for target 'dev' failed
make: *** [dev] Error 1

--- a/ansible/roles/securethenews-deploy/tasks/main.yml
+++ b/ansible/roles/securethenews-deploy/tasks/main.yml
@@ -51,6 +51,7 @@
   become: yes
   pip:
     name: git+git://github.com/dhs-ncats/pshtt@7362a8bdabe7190a933e4c5e18561ba9d07dc561#egg=pshtt
+    executable: pip2.7
 
 - name: Install teamocil
   become: yes

